packages/plugin-node-modules/sources/NodeModulesLinker.ts

```diff
@@ -416,12 +416,12 @@ async function persistNodeModules(preinstallState: NodeModulesLocatorMap | null,
   const addQueue: Promise<void>[] = [];
   const addModule = async ({srcDir, dstDir, linkType, keepNodeModules}: {srcDir: PortablePath, dstDir: PortablePath, linkType: LinkType, keepNodeModules: boolean}) => {
     addQueue.push(limit(async () => {
-      // Soft links to themselves are used to denote workspace packages, we
-      // should just ignore them
-      if (linkType === LinkType.SOFT && srcDir === dstDir)
-        return;
-
       try {
+        // Soft links to themselves are used to denote workspace packages, we
+        // should just ignore them
+        if (linkType === LinkType.SOFT && srcDir === dstDir)
+          return;
```

Comment:
Aaaah - my bad, that's probably on me. I try to make `try` statements as small as possible but I missed the side effect on the `finally`.